tracing-core/src/dispatch.rs

```diff
@@ -404,12 +408,12 @@ where
                     // don't redo this call on the next check
                     *default = get_global().clone();
                 }
-                return f(&*default);
+                return f.take().unwrap()(&*default);
             }
 
-            f(&Dispatch::none())
+            f.take().unwrap()(&Dispatch::none())
         })
-        .unwrap_or_else(|_| f(&Dispatch::none()))
+        .unwrap_or_else(|_| f.take().unwrap()(&Dispatch::none()))
```

Comment:
nit, take it or leave it: probably worth adding an `expect` message to these `unwrap`s explaining why they should never be triggered? not because we expect them to ever be triggered, but fore the benefit of someone reading through this file later and noticing a bunch of `unwrap`s...

Reply:
Will do tomorrow. Thanks!

Reply:
Addressed.